src/Microsoft.ML.ImageAnalytics/MLImage.cs

```diff
@@ -0,0 +1,441 @@
+
+            using SKCanvas canvas = new SKCanvas(dst);
+            using SKPaint paint = new SKPaint() { FilterQuality = SKFilterQuality.High };
+
+            canvas.DrawBitmap(_image, srcRect, destRect, paint);
+
+            return dst;
+        }
+
+        // This matrix get multiplied to every pixel matrix [R G B A W] to average the colors values to get the grayscale effect.
+        private static readonly SKColorFilter _grayscaleColorMatrix = SKColorFilter.CreateColorMatrix(new float[]
+                                                                        {
+                                                                            0.3f, 0.59f, 0.11f, 0, 0,
+                                                                            0.3f, 0.59f, 0.11f, 0, 0,
+                                                                            0.3f, 0.59f, 0.11f, 0, 0,
+                                                                            0,    0,     0,     1, 0
+                                                                        });
+
+        internal MLImage CloneWithGrayscale()
+        {
+            ThrowInvalidOperationExceptionIfDisposed();
+
+            SKBitmap dst = new SKBitmap(_image.Width, _image.Height, isOpaque: true);
+            using SKPaint paint = new SKPaint()
+            {
+                ColorFilter = _grayscaleColorMatrix,
+                FilterQuality = SKFilterQuality.High
+            };
+
+            SKBitmap destBitmap = new SKBitmap(_image.Width, _image.Height, isOpaque: true);
+            using SKCanvas canvas = new SKCanvas(destBitmap);
+            canvas.DrawBitmap(_image, 0f, 0f, paint: paint);
+            return new MLImage(destBitmap);
+        }
+
+        private static readonly Dictionary<string, SKEncodedImageFormat> _extensionToEncodingFormat = new Dictionary<string, SKEncodedImageFormat>(StringComparer.OrdinalIgnoreCase)
+        {
+            { ".bmp", SKEncodedImageFormat.Bmp },
+            { ".png", SKEncodedImageFormat.Png },
+            { ".jpg", SKEncodedImageFormat.Jpeg },
+            { ".jpeg", SKEncodedImageFormat.Jpeg },
+            { ".gif", SKEncodedImageFormat.Gif },
+            { ".ico", SKEncodedImageFormat.Ico },
+            { ".astc", SKEncodedImageFormat.Astc },
+            { ".avif", SKEncodedImageFormat.Avif },
+            { ".dng", SKEncodedImageFormat.Dng },
+            { ".heif", SKEncodedImageFormat.Heif },
+            { ".ktx", SKEncodedImageFormat.Ktx },
+            { ".pkm", SKEncodedImageFormat.Pkm },
+            { ".wbmp", SKEncodedImageFormat.Wbmp },
+            { ".webp", SKEncodedImageFormat.Webp }
+        };
+
+        private void ThrowInvalidOperationExceptionIfDisposed()
+        {
+            if (_image is null)
+            {
+                throw new InvalidOperationException("Object is disposed.");
+            }
+        }
+    }
+
+    /// <summary>
+    /// The mode to decide how the image should be resized.
```

Comment:
Comment seems wrong.

Reply:
Thanks for the catch. it is copy/paste problem :-)

Reply:
Fixed.